cmds/core/wget/wget_test.go

```diff
@@ -40,63 +44,122 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	{
-		name:    "basic",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	{
-		name:    "ipv4",
-		flags:   []string{},
-		url:     "http://127.0.0.1:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	// TODO: CircleCI does not support ipv6
-	// {
-	// 	name:    "ipv6",
-	// 	flags:   []string{},
-	// 	url:     "http://[::1]:%[1]d/200",
-	// 	content: content,
-	// 	retCode: 0,
-	// },
-	{
-		name:    "redirect",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/302",
-		content: "",
-		retCode: 0,
-	},
-	{
-		name:    "4xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/404",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "5xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/500",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "no server",
-		flags:   []string{},
-		url:     "http://localhost:%[2]d/200",
-		content: "",
-		retCode: 1,
-	},
+// TestWget implements a table-driven test.
+func TestWget(t *testing.T) {
+	srv := httptest.NewServer(handler{})
+	defer srv.Close()
+
+	// os.Getwd is needed to test default output path
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// return back to initial dir
+	defer os.Chdir(dir)
+
+	var tests = []struct {
+		name        string
+		url         string // in
+		wantContent string // out
+		outputPath  string
+		wantErr     error
+	}{
+		{
+			name:        "ipv4",
+			url:         fmt.Sprintf("%s/200", srv.URL),
+			wantContent: content,
+			outputPath:  "basic",
+			wantErr:     nil,
+		},
+		{
+			name:        "index.html-1",
+			url:         fmt.Sprintf("%s/", srv.URL),
+			wantContent: content,
+			outputPath:  "",
+			wantErr:     nil,
+		},
+		{
+			name:        "index.html-2",
+			url:         srv.URL,
+			wantContent: content,
+			outputPath:  "",
+			wantErr:     nil,
+		},
+		{
+			name:        "localhost",
+			url:         strings.Replace(srv.URL, "127.0.0.1", "localhost", 1) + "/200",
+			wantContent: content,
+			outputPath:  "ipv4",
+			wantErr:     nil,
+		},
+		// TODO: CircleCI does not support ipv6
+		// {
+		// 	name:    "ipv6",
+		// 	flags:   []string{},
+		// 	url:     "http://[::1]:%[1]d/200",
+		// 	content: content,
+		// 	retCode: 0,
+		// },
+		{
+			name:       "redirect",
+			url:        fmt.Sprintf("%s/302", srv.URL),
+			outputPath: "redirect",
+			wantErr:    nil,
+		},
+		{
+			name:    "4xx error",
+			url:     fmt.Sprintf("%s/404", srv.URL),
+			wantErr: curl.ErrStatusNotOk,
+		},
+		{
+			name:    "5xx error",
+			url:     fmt.Sprintf("%s/500", srv.URL),
+			wantErr: curl.ErrStatusNotOk,
+		},
+		{
+			name:    "empty url",
+			url:     "",
+			wantErr: errEmptyURL,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir := t.TempDir()
+			err := os.Chdir(tempDir)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if tt.outputPath != "" {
+				tt.outputPath = filepath.Join(tempDir, tt.outputPath)
```

Comment:
The test now `ChDir`s into tempDir so there's no need to do this `filepath.Join` right?

Reply:
yep